### ISSUE TITLE
Cannot Start ProB2 UI

### ISSUE THREAD

**Issue by rutenkolk** [hhu-stups/prob-issues#105]
OS: WIN11

Tested Versions: 
* ❌ Snapshot 1.2.0 obtained here: `https://www3.hhu.de/stups/downloads/prob2/snapshot/ProB%202%20UI-1.2.0.exe`
* ❌ MultiJar version 1.1.0 & 1.2.0
* ✅ Version 1.1.0 with bundled JRE does **not** display this error and works correctly.

Further System Details as per `systeminfo` :

```
Betriebssystemname:                            Microsoft Windows 11 Pro
Betriebssystemversion:                         10.0.22000 Nicht zutreffend Build 22000
Betriebssystemhersteller:                      Microsoft Corporation
Betriebssystemkonfiguration:                   Eigenständige Arbeitsstation
Betriebssystem-Buildtyp:                       Multiprocessor Free
Systemhersteller:                              To Be Filled By O.E.M.
Systemmodell:                                  To Be Filled By O.E.M.
Systemtyp:                                     x64-based PC
Prozessor(en):                                 1 Prozessor(en) installiert.
                                               [01]: Intel64 Family 6 Model 158 Stepping 12 GenuineIntel ~3601 MHz
BIOS-Version:                                  American Megatrends Inc. P4.30, 12.08.2019
Windows-Verzeichnis:                           C:\WINDOWS
System-Verzeichnis:                            C:\WINDOWS\system32
Systemgebietsschema:                           de;Deutsch (Deutschland)
Eingabegebietsschema:                          de;Deutsch (Deutschland)
Zeitzone:                                      (UTC+01:00) Amsterdam, Berlin, Bern, Rom, Stockholm, Wien
Gesamter physischer Speicher:                  32.681 MB
Verfügbarer physischer Speicher:               21.932 MB
Virtueller Arbeitsspeicher: Maximale Größe:    37.545 MB
Virtueller Arbeitsspeicher: Verfügbar:         23.126 MB
Virtueller Arbeitsspeicher: Zurzeit verwendet: 14.419 MB
Auslagerungsdateipfad(e):                      C:\pagefile.sys
Domäne:                                        WORKGROUP
Hotfix(es):                                    4 Hotfix(e) installiert.
                                               [01]: KB5008880
                                               [02]: KB5004567
                                               [03]: KB5009566
                                               [04]: KB5007414
```

Starting Prob2 as a GUI Application simply hangs and the application never exits on it's own. There is no feedback to the user and the process needs to be terminated manually.

Using the console version, an exception is being thrown:

```
PS C:\Program Files\ProB2UI_SNAPSHOT> & '.\ProB 2 UI (with console).exe'
Exception in thread "WindowsNativeRunloopThread" java.lang.NoSuchMethodError: <init>
        at com.sun.glass.ui.win.WinApplication.staticScreen_getScreens(Native Method)
        at com.sun.glass.ui.Screen.initScreens(Screen.java:412)
        at com.sun.glass.ui.Application.lambda$run$1(Application.java:152)
        at com.sun.glass.ui.win.WinApplication._runLoop(Native Method)
        at com.sun.glass.ui.win.WinApplication.lambda$runLoop$3(WinApplication.java:174)
        at java.base/java.lang.Thread.run(Unknown Source)
Exception in thread "WindowsNativeRunloopThread" java.lang.NoSuchMethodError: <init>
        at com.sun.glass.ui.win.WinApplication.staticScreen_getScreens(Native Method)
        at com.sun.glass.ui.Screen.initScreens(Screen.java:412)
        at com.sun.glass.ui.Application.lambda$run$1(Application.java:152)
        at com.sun.glass.ui.win.WinApplication._runLoop(Native Method)
        at com.sun.glass.ui.win.WinApplication.lambda$runLoop$3(WinApplication.java:174)
        at java.base/java.lang.Thread.run(Unknown Source)
Exception in thread "WindowsNativeRunloopThread" java.lang.NoSuchMethodError: <init>
        at com.sun.glass.ui.win.WinApplication.staticScreen_getScreens(Native Method)
        at com.sun.glass.ui.Screen.initScreens(Screen.java:412)
        at com.sun.glass.ui.Application.lambda$run$1(Application.java:152)
        at com.sun.glass.ui.win.WinApplication._runLoop(Native Method)
        at com.sun.glass.ui.win.WinApplication.lambda$runLoop$3(WinApplication.java:174)
        at java.base/java.lang.Thread.run(Unknown Source)
Exception in thread "JavaFX Application Thread" java.lang.NullPointerException: Cannot invoke "com.sun.glass.ui.Screen.getNativeScreen()" because "<parameter1>" is null
        at com.sun.prism.d3d.D3DPipeline.getAdapterOrdinal(D3DPipeline.java:205)
        at com.sun.javafx.tk.quantum.QuantumToolkit.assignScreensAdapters(QuantumToolkit.java:695)
        at com.sun.javafx.tk.quantum.QuantumToolkit.runToolkit(QuantumToolkit.java:313)
        at com.sun.javafx.tk.quantum.QuantumToolkit.lambda$startup$10(QuantumToolkit.java:258)
        at com.sun.glass.ui.Application.lambda$run$1(Application.java:153)
        at com.sun.glass.ui.win.WinApplication._runLoop(Native Method)
        at com.sun.glass.ui.win.WinApplication.lambda$runLoop$3(WinApplication.java:174)
        at java.base/java.lang.Thread.run(Unknown Source)
Exception in thread "Glass/Prism Shutdown Hook" java.lang.NullPointerException: Cannot invoke "com.sun.glass.ui.Timer.stop()" because "this.pulseTimer" is null
        at com.sun.javafx.tk.quantum.QuantumToolkit.dispose(QuantumToolkit.java:798)
        at com.sun.javafx.tk.quantum.QuantumToolkit$1.run(QuantumToolkit.java:231)
PS C:\Program Files\ProB2UI_SNAPSHOT>
```

The exact same Exception/Stacktrace is being thrown when trying to start ProB2UI via `java -jar`.
The java Version used does not seem to change that. Tested Java versions: Java 17-lts temurin (adoptium) & OpenJDK 11

**Comment by dgelessus** [comment]
Still cannot reproduce this, sorry... Both the installer with bundled JRE and multi-jar with standalone JDK (Adoptium) work fine for me with the current snapshot.

My machine is on the same Windows 11 build. It's Windows 11 Home and not Pro, and `systeminfo` shows a slightly different list of hotfixes, but I doubt that makes a difference...

**Comment by rutenkolk** [comment]
Ok I googled a bit and found this:
https://github.com/binjr/binjr/issues/111#issuecomment-827725723

So first I tested if my JAVA_HOME was set correctly. It is, as I installed my jdk17 with scoop.

Then I looked if glass.dll is somewhere on my PATH... and it is! There is the standard oracle JRE 8 installed on my Windows System which doesn't even get used since the JDK 17 binary is also on the PATH, but with higher Priority.

So according to this bugreport: 
https://bugs.openjdk.java.net/browse/JDK-8246796
or it's relative bug report
https://bugs.openjdk.java.net/browse/JDK-8223746

this is a known issue without a direct fix from JavaFX (yet).

What I tried as a workaround is simply extracting the glass.dll from the jar in `app/prob2-ui-1.2.0-SNAPSHOT-win.jar` and copying it next to the `.exe` so it gets a higher loading priority than the one on PATH.

This actually works(!), but it's quite a weird hack. But I don't think requiring windows users to not install java is the way to go (since jre8 ist still the default download from oracle).
